client/src/components/CalculateMenu/CalculateMenu.js

```diff
@@ -12,33 +14,39 @@ export const CalculateMenu = ({
   setDirectionsResponse,
   setIsCalculateRoute,
 }) => {
-  const destiantionRef = useRef();
-  const originRef = useRef();
-
   const [distance, setDistance] = useState("");
   const [duration, setDuration] = useState("");
 
+  const [origin, setOrigin] = useState(`${centerAddress}`);
+  const [destination, setDestintion] = useState(`${selectedMarker.address}`);
+
   const makeRoute = () => {
     window.location.href = `https://www.google.com/maps/dir/${center.lat},${center.lng}/${selectedMarker.location.lat},${selectedMarker.location.lng}`;
   };
 
+  useEffect(() => {
+    setOrigin(`${centerAddress}`);
+    setDestintion(`${selectedMarker.address}`);
+  }, [centerAddress, selectedMarker.address]);
+
   function clearRoute() {
-    originRef.current.value = "";
-    destiantionRef.current.value = "";
+    setOrigin("");
+    setDestintion("");
+
     setDirectionsResponse(null);
     setDistance("");
     setDuration("");
   }
 
   async function calculateRoute() {
-    if (originRef.current.value === "" || destiantionRef.current.value === "") {
+    if (origin === "" || destination === "") {
       return;
     }
     // eslint-disable-next-line no-undef
     const directionsService = new google.maps.DirectionsService();
     const results = await directionsService.route({
-      origin: originRef.current.value,
-      destination: destiantionRef.current.value,
+      origin: origin,
```

Comment:
if object has same key and value, like `origin: origin`, we can simplify this with just `origin`